src/services/balancer/subgraph/entities/pools/index.ts

```diff
@@ -40,13 +40,17 @@ export default class Pools {
     attrs = {}
   ): Promise<DecoratedPool[]> {
     attrs = { ...attrs, __aliasFor: 'pools' };
-    const block = { number: await this.timeTravelBlock(period) };
-    const query = {
-      currentPools: this.query(args, attrs).pools,
-      pastPools: this.query({ ...args, block }, attrs).pools
+    const currentPoolsQuery = {
+      currentPools: this.query(args, attrs).pools
     };
+    const { currentPools } = await this.service.client.get(currentPoolsQuery);
 
-    const { currentPools, pastPools } = await this.service.client.get(query);
+    const block = { number: await this.timeTravelBlock(period) };
```

Comment:
`currentPools` and `block` can be part of a `promise.all` so we could benefit from parallelism